crates/bevy_render/src/globals.rs

```diff
@@ -0,0 +1,74 @@
+use crate::{
+    extract_resource::ExtractResource,
+    render_resource::{ShaderType, UniformBuffer},
+    renderer::{RenderDevice, RenderQueue},
+    Extract, RenderApp, RenderStage,
+};
+use bevy_app::{App, Plugin};
+use bevy_ecs::prelude::*;
+use bevy_log::info;
+use bevy_reflect::Reflect;
+use bevy_time::Time;
+
+#[derive(Default, Resource)]
+struct FrameCount(u32);
+
+pub struct GlobalsPlugin;
+
+impl Plugin for GlobalsPlugin {
+    fn build(&self, app: &mut App) {
+        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app
+                .init_resource::<GlobalsBuffer>()
+                .init_resource::<Time>()
+                .init_resource::<FrameCount>()
+                .add_system_to_stage(RenderStage::Extract, extract_time)
+                .add_system_to_stage(RenderStage::Prepare, prepare_globals_buffer);
+        }
+    }
+}
+
+fn extract_time(mut commands: Commands, time: Extract<Res<Time>>) {
+    commands.insert_resource(time.clone());
+}
+
+/// Contains global values useful when writing shaders.
+/// Currently only contains values related to time.
+#[derive(Default, Clone, Resource, ExtractResource, Reflect, ShaderType)]
+#[reflect(Resource)]
+pub struct GlobalsUniform {
+    /// The time since startup in seconds
+    time: f32,
+    /// The duration of the last frame in seconds
+    delta_time: f32,
+    /// Frame count since the start of the app.
+    frame_count: u32,
+}
+
+/// The buffer containing the [`GlobalsUniform`]
+#[derive(Resource, Default)]
+pub struct GlobalsBuffer {
+    pub buffer: UniformBuffer<GlobalsUniform>,
+}
+
+fn prepare_globals_buffer(
+    render_device: Res<RenderDevice>,
+    render_queue: Res<RenderQueue>,
+    mut globals_buffer: ResMut<GlobalsBuffer>,
+    time: Res<Time>,
+    mut frame_count: Local<u32>,
```

Comment:
I feel like maybe having the frame count in the main world is useful for something so maybe add it as a resource there and extract it.

Reply:
My issue is that the frame count is available as a diagnostic if you use the `FrameTimeDiagnosticsPlugin`, but since it's not included by default I can't rely on it being present. I'm not sure if we should push people to use this plugin when they want the frame count or if the frame count should just be a resource that's always present and the diagnostic plugin just uses it directly.

Reply:
The latter is what I was thinking. I haven’t thought about how much time that many frames would represent… if i calculated correctly, just less than a year of 144Hz as that comes out to just over 345 days of frames. I guess similar to time we could/should use higher precision (ie u64) for the resource and then wrap into u32 for usage with that precision. u64 would store about 4 billion years of 144Hz frames before wrapping. :) Not quite the age of the universe but if I recall correctly maybe the age of the sun ish.

Reply:
Right, makes sense, I'll do it like that then.

Reply:
I added the FrameCount reousrce to bevy_core, to make it available from anywhere, but the plugin in charge of updating it is in bevy_render. The idea being that this is the frame count so if bevy isn't rendering anything it shouldn't update the count.

Reply:
Hmmmmm. When does FrameTimeDiagnosticsPlugin update the frame number currently? I’d do the same as that

Reply:
Currently it just updates everytime the system is called which is also what I do in this PR